tilequeue/command.py

```diff
@@ -271,32 +271,60 @@ def tilequeue_drain(cfg, peripherals):
     logger.info('Removed %d messages' % n)
 
 
-def explode_and_intersect(coord_ints, tiles_of_interest, until=0):
+def explode_and_intersect(
+        coord_ints, tiles_of_interest, intersect_metrics, until=0):
+
     next_coord_ints = coord_ints
     coord_ints_at_parent_zoom = set()
+
+    # to capture metrics
+    total = 0
+    hits = 0
+    misses = 0
+
     while True:
+
         for coord_int in next_coord_ints:
+
+            total += 1
             if coord_int in tiles_of_interest:
+                hits += 1
                 yield coord_int
+            else:
+                misses += 1
+
             zoom = zoom_mask & coord_int
             if zoom > until:
                 parent_coord_int = coord_int_zoom_up(coord_int)
                 coord_ints_at_parent_zoom.add(parent_coord_int)
+
         if not coord_ints_at_parent_zoom:
+            intersect_metrics['total'] = total
```

Comment:
Modifying a `dict` that's passed in to the function is not very pythonic and could lead to unintended side-effects. It looks like the `coord_ints` thing we're passing in here is just a `set` of `set`s (i.e. not a generator or iterator) so we don't save anything by using a generator here. Can we just return the list of `coord_int` instead of `yield`ing them individually and also return the stats object?